Deal with ncurses in build system: detect, or don't use...

The following build failure can happen on Ubuntu (and in a similar fashion on other systems), despite `libreadline-dev` being installed:
```
$ ./configure --with-readline
...
checking readline/readline.h usability... yes
checking readline/readline.h presence... yes
checking for readline/readline.h... yes
checking for rl_ding in -lreadline... no
configure: error: in `/mnt/c/Users/caj/files/reps/gap/gap':
configure: error: --with-readline was given, but test for readline failed
```
The problem becomes clear when looking at `config.log`:
```
...
configure:18014: checking for readline/readline.h
configure:18014: result: yes
configure:18017: checking for rl_ding in -lreadline
configure:18042: gcc -o conftest -g -O2     conftest.c -lreadline -lncurses  >&5
/usr/bin/ld: cannot find -lncurses
...
```

So the problem was/is caused by us always adding `-lncurses` to the linker flags, but `libncurses5-dev` may not be installed.

This is very annoying to debug for a user, as there is no hint about `ncurses` in the terminal output.

I see two ways to resolve this:

1. Add an explicit test for `-lncurses` before the readline tests; this way, we fail in a more appropriate place, giving the user a hint as to what goes wrong. In addition, I would suggest that we add somewhere in `INSTALL.md`, `README.buildsys.md` or `README.md` a hint about packages to install on "typical" systems (say Ubuntu/Debian, Fedora, macOS with Homebrew/Fink/MacPorts, and happily others (e.g. @markuspf hopefully would add DragonFly BSD). Of course this should also list the "required" GMP package(s). And possibly even mention other optional dependencies useful for packages (?)

2. Stop adding `-lncurses` there. I am not sure whether it is needed. While, I think readline does use ncurses (or at least optional does so), on my Mac, I can drop the `-lncurses` and everything is fine.

--- a/configure.ac
+++ b/configure.ac
@@ -328,7 +328,7 @@ dnl Now check if we can find GNU readline. We go to some extra efforts to
 dnl ensure it is GNU readline, and not e.g. BSD editline wrappers for
 dnl readline, which do not suffice for GAP.
 AS_IF([test "x$with_readline" != xno],[
-  READLINE_LIBS="-lreadline -lncurses"
+  READLINE_LIBS="-lreadline"
   AS_CASE([x"$with_readline"],
     [xyes|x],[
       READLINE_CPPFLAGS=
@@ -342,7 +342,7 @@ AS_IF([test "x$with_readline" != xno],[
 
   dnl We check for the symbol rl_ding to distinguish GNU readline
   dnl from other readline implementation.
-  AX_CHECK_LIBRARY([READLINE], [readline/readline.h], [readline], [rl_ding], [], [], [-lncurses])
+  AX_CHECK_LIBRARY([READLINE], [readline/readline.h], [readline], [rl_ding], [], [], [])
 
   AS_IF([test $ax_cv_have_READLINE = yes],[
     AC_DEFINE([HAVE_LIBREADLINE], [1], [Define if you have libreadline])

